src/base/bittorrent/sessionimpl.cpp

```diff
@@ -2892,26 +2892,44 @@ void SessionImpl::saveResumeData()
         ++m_numResumeData;
     }
 
+    const lt::seconds waitTime {30};
+    QElapsedTimer timer;
+    timer.start();
+
     while (m_numResumeData > 0)
     {
-        const std::vector<lt::alert *> alerts = getPendingAlerts(lt::seconds {30});
-        if (alerts.empty())
-        {
-            LogMsg(tr("Aborted saving resume data. Number of outstanding torrents: %1").arg(QString::number(m_numResumeData))
-                , Log::CRITICAL);
-            break;
-        }
+        const std::vector<lt::alert *> alerts = getPendingAlerts(waitTime);
 
+        bool hasWantedAlert = false;
         for (const lt::alert *a : alerts)
         {
             switch (a->type())
             {
             case lt::save_resume_data_failed_alert::alert_type:
+                hasWantedAlert = true;
                 --m_numResumeData;
                 break;
             case lt::save_resume_data_alert::alert_type:
+                hasWantedAlert = true;
                 dispatchTorrentAlert(static_cast<const lt::torrent_alert *>(a));
                 break;
+            default:
+                // skip over other alert types
```

Comment:
Aren't they skipped without this `default` case? Or is it just to put comment inside?

Reply:
>Or is it just to put comment inside?

This. IMO `switch()` should cover all possible cases.

Reply:
> IMO `switch()` should cover all possible cases.

But all you're doing is instead of implicitly skipping all but two cases, you're doing it explicitly. Is that what you mean by "cover all possible cases"?

Reply:
>Is that what you mean by "cover all possible cases"?

For those are skipped, they should be mentioned explicitly (here is done in the comment).
So now 2 cases are explicitly handled and others are explicitly ignored/skipped and therefore every possible case are written out (none are left behind the scenes).

Reply:
> For those are skipped, they should be mentioned explicitly

IMO, they **can** be mentioned explicitly.
>here is done in the comment

And, IMO, the comment is enough w/o `default` case since it is only for readers, it can't improve anything else in this case.

Reply:
>IMO, they can be mentioned explicitly.

It depends. https://stackoverflow.com/a/62467351
Here is `switch(int)` so it should be better to have a default case.

Reply:
>https://stackoverflow.com/a/62467351

The precondition for the correctness of this method is the use of a static analyzer (as for the so-called "exhaustive" cases). If you don't rely on static analyzer you have to add `delault` to guard from further changes of switched enumeration.

As for the "non-exhaustive" cases, I do not agree with the author of the comment. On the contrary, if you intend to ignore all cases except those that you explicitly handle, there is no point in having a `default` label (well, except to put a comment there, but you can still have such a comnent without `default` label).

Reply:
Okay, I don't care about this `default` label too much. Let's focus on the essence of this PR.